pkg/controller/restore_finalizer_controller.go

```diff
@@ -231,8 +232,15 @@ func (r *restoreFinalizerReconciler) finishProcessing(restorePhase velerov1api.R
 		r.metrics.RegisterRestoreSuccess(restore.Spec.ScheduleName)
 	}
 	restore.Status.CompletionTimestamp = &metav1.Time{Time: r.clock.Now()}
-
-	return kubeutil.PatchResource(original, restore, r.Client)
+	// retry using DefaultBackoff to resolve connection refused error that may occur when the server is under heavy load
+	// TODO: consider using a more specific error type to retry, for now, we retry on all errors
+	// specific errors:
+	// - connection refused: https://pkg.go.dev/syscall#:~:text=Errno(0x67)-,ECONNREFUSED,-%3D%20Errno(0x6f
+	return retry.OnError(retry.DefaultBackoff, func(err error) bool {
```

Comment:
double check my math if this backoff is enough time to solve linked issue.
https://pkg.go.dev/k8s.io/client-go@v0.29.0/util/retry#pkg-variables
```
var DefaultBackoff = wait.Backoff{
	Steps:    4,
	Duration: 10 * time.Millisecond,
	Factor:   5.0,
	Jitter:   0.1,
}
```